hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -487,7 +487,7 @@ private int getBlockCount(List<ExtendedBlocklet> blocklets) {
     List<ExtendedBlocklet> prunedBlocklets = null;
     // This is to log the event, so user will know what is happening by seeing logs.
     LOG.info("Started block pruning ...");
-    if (carbonTable.isTransactionalTable()) {
+    if (carbonTable.isTransactionalTable() && !carbonTable.hasColumnDrift()) {
```

Comment:
I feel it is better check this `hasColumnDrift` on individual segement properties rather than CarbonTable.  CarbonTable is  always updated with latest schema. So once the compaction of old segements happen then all the old segments are updated with latest schema then this `hasColumnDrift` does not make sense.

In the below code going to else block is basically costly operation as it needs to resolve the filter expression per each segment. So I feel it should be better if we have one more check to compare whether the current schema and segment schema has different columnDrift

Reply:
fixed
if the segment has no column drift, it will reuse filterResolver.
if the segment has column drift, it will use filterExpression to build filterResolver for this segment